A cursory glance at the data suggests the length of the bridge is mostly below 3000. There are only a few points that lie above or equal to this value at a glance.

In [1]:
import pandas as pd
df=pd.read_csv("bridges.csv")

df.describe()

,location,erected,length,lanes
count,107.000000,108.000000,81.000000,92.000000
mean,25.978505,1905.314815,1567.469136,2.630435
std,13.665871,37.173664,747.491523,1.164657
min,1.000000,1818.000000,804.000000,1.000000
25%,15.500000,1884.000000,1000.000000,2.000000
50%,27.000000,1903.000000,1300.000000,2.000000
75%,37.500000,1928.000000,2000.000000,4.000000
max,52.000000,1986.000000,4558.000000,6.000000


In [2]:
df[df["length"]>=3000]

,Id,river,location,erected,purpose,length,lanes,clear-g,t-or-d,material,span,rel-l,type
31,E34,O,41.0,1888,RR,4558.0,2.0,G,THROUGH,STEEL,LONG,F,SIMPLE-T
44,E46,A,37.0,1897,RR,4000.0,2.0,G,DECK,STEEL,LONG,F,SIMPLE-T
52,E87,A,35.0,1903,RR,3000.0,2.0,G,THROUGH,STEEL,MEDIUM,S,SIMPLE-T
104,E91,O,44.0,1975,HIGHWAY,3756.0,6.0,G,THROUGH,STEEL,LONG,F,ARCH


Of the 108 total rows of data, 4 of them are above or equal to 3000, however, 27 are not entered into the data set. This makes the 4 above or equal to 3000 have a fraction of 4/81, roughly 5%. It is a small amount of data that should not be removed or replaced since the mean will substantially decrease from having these 4 extreme values removed. Ultimately, none of these lengths should be removed or replaced.

Repo Address: https://github.com/KyeSPu/cs82a-portfolio